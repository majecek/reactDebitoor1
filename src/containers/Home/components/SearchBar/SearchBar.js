import React, { Component } from 'react'
import PropTypes from 'prop-types'
import TextField from 'material-ui/TextField'

class SearchBar extends Component {
  constructor (props) {
    super(props)
    this.state = {
      term: ''
    }
  }

  onInputChange = (event) => {
    this.setState({ term: event.target.value })
  }

  onFormSubmit = (event) => {
    event.preventDefault()
    this.props.getRepositoryOrderedByStars(this.state.term)
  }

  render () {
    return (
      <div>
        <form onSubmit={this.onFormSubmit}>
          <TextField
            id="searchfield"
            hintText="Search for repo"
            value={this.state.term}
            onChange={this.onInputChange}
          />

          <span className="input-group-btn">
            <button type="submit" className="btn btn-secondary">Submit</button>
          </span>
        </form>
      </div>
    )
  }
}

SearchBar.propTypes = {
  getRepositoryOrderedByStars: PropTypes.func
}

export default SearchBar