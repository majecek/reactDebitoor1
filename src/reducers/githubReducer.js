import axios from 'axios'

// ACTION TYPES
const REPOS_ORDERED_BY_STARS = 'REPOS_ORDERED_BY_STARS'
const PULL_REQUESTS = 'PULL_REQUESTS'
const URL_REPOS_ORDERED_BY_STARS_PREFIX = 'https://api.github.com/search/repositories?q='
const URL_REPOS_ORDERED_BY_STARS_POSTFIX = 'stars:%3E1&sort=stars&order=desc'
const URL_PULL_REQUESTS = 'https://api.github.com/repos/'
const initialState = {}
let searchTerm = ''

// REDUCERS
export default function (state = initialState, action = {}) {
  switch (action.type) {

    case REPOS_ORDERED_BY_STARS:
      return {
        ...state,
        repos: action.payload.data,
      }

    case PULL_REQUESTS:
      return {
        ...state,
        pullRequests: action.payload.data
      }

    default:
      return state
  }
}

// ACTIONS
export function getRepositoryOrderedByStars (term = searchTerm) {
  let URL = ''
  if (term) {
    URL = `${URL_REPOS_ORDERED_BY_STARS_PREFIX}${term}?${URL_REPOS_ORDERED_BY_STARS_POSTFIX}`
  } else {
    URL = `${URL_REPOS_ORDERED_BY_STARS_PREFIX}${URL_REPOS_ORDERED_BY_STARS_POSTFIX}`
  }
  const request = axios.get(`${URL}`)
  searchTerm = term

  return {
    type: REPOS_ORDERED_BY_STARS,
    payload: request
  }
}

export function getPullRequests (repoUrl) {
  const URL = `${URL_PULL_REQUESTS}${repoUrl}/pulls`
  const request = axios.get(`${URL}`)

  return {
    type: PULL_REQUESTS,
    payload: request
  }
}
