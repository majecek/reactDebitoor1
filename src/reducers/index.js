import { combineReducers } from 'redux'
import GithubReducer from './githubReducer'

const rootReducer = combineReducers({
  github: GithubReducer
})

export default rootReducer
