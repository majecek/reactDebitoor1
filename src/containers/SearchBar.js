import React, { Component } from 'react'
import TextField from 'material-ui/TextField'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import { getRepositoryOrderedByStars } from './../reducers/githubReducer'

class SearchBar extends Component {

  constructor (props) {
    super(props)

    this.state = {term: ''}

    this.onInputChange = this.onInputChange.bind(this)
    this.onFormSubmit = this.onFormSubmit.bind(this)
  }

  onInputChange (event) {
    this.setState({term: event.target.value})
  }

  onFormSubmit (event) {
    event.preventDefault()

    this.props.getRepositoryOrderedByStars(this.state.term)
  }

  render () {
    return (
      <div>
        <form onSubmit={this.onFormSubmit}>
          <TextField
            hintText="Search for repo"
            value={this.state.term}
            onChange={this.onInputChange}
          />

          <span className="input-group-btn">
            <button type="submit" className="btn btn-secondary">Submit</button>
          </span>
        </form>
      </div>
    )
  }
}

SearchBar.propTypes = {}
SearchBar.defaultProps = {}

function mapDispatchToProps (dispatch) {
  return bindActionCreators({getRepositoryOrderedByStars}, dispatch)
}

export default connect(null, mapDispatchToProps)(SearchBar)
