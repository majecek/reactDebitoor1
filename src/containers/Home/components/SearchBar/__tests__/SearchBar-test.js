import React from 'react'
import renderer from 'react-test-renderer'
import SearchBar from '../SearchBar'
import { getRepositoryOrderedByStars } from '../../../../../reducers/githubReducer'
import baseTheme from 'material-ui/styles/baseThemes/lightBaseTheme';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
const theme = getMuiTheme(baseTheme)

describe('SearchBar', () => {
  const result = renderer.create(
    <MuiThemeProvider muiTheme={theme}>
      <SearchBar />
    </MuiThemeProvider>).toJSON()

  it('should render full html structure correctly', () => {
    expect(result).toMatchSnapshot()
  })
})