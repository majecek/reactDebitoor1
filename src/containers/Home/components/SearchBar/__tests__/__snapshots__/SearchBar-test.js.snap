exports[`SearchBar should render full html structure correctly 1`] = `
<div>
  <form
    onSubmit={[Function]}>
    <div
      className={undefined}
      style={
        Object {
          "backgroundColor": "transparent",
          "cursor": "auto",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": 16,
          "height": 48,
          "lineHeight": "24px",
          "muiPrepared": true,
          "position": "relative",
          "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "width": 256,
        }
      }>
      <div
        style={
          Object {
            "bottom": 12,
            "color": "rgba(0, 0, 0, 0.3)",
            "muiPrepared": true,
            "opacity": 1,
            "position": "absolute",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          }
        }>
        Search for repo
      </div>
      <input
        disabled={false}
        id="searchfield"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0,0,0,0)",
            "appearance": "textfield",
            "backgroundColor": "rgba(0,0,0,0)",
            "border": "none",
            "color": "rgba(0, 0, 0, 0.87)",
            "cursor": "inherit",
            "font": "inherit",
            "height": "100%",
            "muiPrepared": true,
            "outline": "none",
            "padding": 0,
            "position": "relative",
            "width": "100%",
          }
        }
        type="text"
        value="" />
      <div>
        <hr
          aria-hidden="true"
          style={
            Object {
              "borderBottom": "solid 1px",
              "borderColor": "#e0e0e0",
              "borderLeft": "none",
              "borderRight": "none",
              "borderTop": "none",
              "bottom": 8,
              "boxSizing": "content-box",
              "margin": 0,
              "muiPrepared": true,
              "position": "absolute",
              "width": "100%",
            }
          } />
        <hr
          aria-hidden="true"
          style={
            Object {
              "borderBottom": "solid 2px",
              "borderColor": "#00bcd4",
              "borderLeft": "none",
              "borderRight": "none",
              "borderTop": "none",
              "bottom": 8,
              "boxSizing": "content-box",
              "margin": 0,
              "muiPrepared": true,
              "position": "absolute",
              "transform": "scaleX(0)",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "width": "100%",
            }
          } />
      </div>
    </div>
    <span
      className="input-group-btn">
      <button
        className="btn btn-secondary"
        type="submit">
        Submit
      </button>
    </span>
  </form>
</div>
`;
